docs/reference/expertfeatures.rst

```diff
@@ -0,0 +1,123 @@
+Inmanta Expert Features
+=======================
+
+This page documents the set of expert features for the Inmanta application and extensions.
+
+.. attention::
+
+        "With great power comes great responsibility"
+
+    These features have to be used with extreme caution. Make sure you are fully aware of what they do before using
+    them. These feature allow you to bypass safety checks that the product offers and/or that were put in place by
+    the model developers.
+
+Compiler expert features
+########################
+
+Allow handler code update during a partial compile
+--------------------------------------------------
+
+
+    By default, updating handler code is only allowed during full compiles. A
+    safety check is performed during partial compiles to make sure that inmanta
+    module code remains consistent across partial versions.
+
+    If a full compile cannot be performed and handler code update is necessary,
+    it is possible to disable the version consistency check and to allow handler
+    code update in a manual export as follows:
+
+
+
+    .. danger::
+        Make sure the updated handler code is compatible with all resource
+        schemas it might have to process. Uploading incompatible handler code can lead
+        to very hard to diagnose issues.
+
+    .. admonition::
+        How-to
+
+        To allow handler code update during a partial compile, pass the
+        :command:`--allow-handler-code-update` option to the :command:`export` command.
+
+        .. code-block:: bash
+
+            inmanta -vv -X export --partial --allow-handler-code-update
+
+
+
+.. only:: iso
+
+    Lsm extension expert features
+    #############################
+
+    To access expert features of the lsm extension, toggle the
+    :inmanta.environment-settings:setting:`enable_lsm_expert_mode` environment setting. This will unlock
+    the features described below as long as the setting is still enabled. Make sure you disable it again when
+    you're done.
+
+    .. danger::
+        A service lifecycle describes how a service can/cannot evolve during its lifespan as a finite state
+        machine. An integral part of The LSM extension is to uphold these constraints by performing safety
+        check on each transition request.
+
+        Using LSM expert features explicitly bypass these safety checks, which means that incorrect usage
+        can lead to a broken desired state and break future compilations.
+
+        These features should never be used, unless you have a very good reason to, you fully understand
+        why the safety checks are there in the first place, and why it is ok to bypass them for your
+        specific use case.
+
+
+
+    Forcefully set a service in a state
+    -----------------------------------
+
+        .. danger::
```

Comment:
Why is this indented? It looks a bit weird in the render (from here till the end)